gammapy/maps/axes.py

```diff
@@ -1797,41 +1818,45 @@ class TimeMapAxis:
         coordinates.  For now only 'lin' is supported.
     """
     node_type = "intervals"
+
     def __init__(self, edges_min, edges_max, reference_time, name="time", interp="lin"):
         self._name = name
 
         edges_min = u.Quantity(edges_min, ndmin=1)
         edges_max = u.Quantity(edges_max, ndmin=1)
 
-        reference_time = Time(reference_time)
-
-        invalid = [_.unit.name for _ in (edges_min, edges_max) if not _.unit.is_equivalent("d")]
-        if len(invalid)>0:
-            raise TypeError(f"TimeAxis edges must be time-like quantities. Got {invalid}")
+        if not edges_min.unit.is_equivalent("s"):
+            raise ValueError(f"Time edges min must have a valid time unit, got {edges_min.unit}")
 
-        # Note: flatten is there to deal with scalr Time objects
-        if not len(edges_min.flatten()) == len(edges_max.flatten()):
-            raise ValueError("Time min and time max must have the same length.")
+        if not edges_max.unit.is_equivalent("s"):
+            raise ValueError(f"Time edges max must have a valid time unit, got {edges_max.unit}")
 
-        if not (np.all(edges_min == np.sort(edges_min))
-                and np.all(edges_max == np.sort(edges_max))):
-            raise ValueError("TimeAxis: edges values must be sorted")
+        if not edges_min.shape == edges_max.shape:
+            raise ValueError("Time min and time max must have the same shape,"
```

Comment:
Here it should be `edges min and edges mas mast have the same shape`

Reply:
Done